Agents/FeatureInfoAgent/code/src/main/java/com/cmclinnovations/featureinfo/kg/TimeHandler.java

```diff
@@ -0,0 +1,396 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.cmclinnovations.featureinfo.FeatureInfoAgent;
+import com.cmclinnovations.featureinfo.Utils;
+import com.cmclinnovations.featureinfo.config.ConfigEndpoint;
+
+import uk.ac.cam.cares.jps.base.query.RemoteStoreClient;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeries;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeriesClient;
+
+/**
+ * This class handles querying a Blazegraph endpoint to determine what measurements
+ * are available, then retrieving these from the PostGreSQL endpoint.
+ */
+public class TimeHandler {
+
+    /**
+     * Logger for reporting info/errors.
+     */
+    private static final Logger LOGGER = LogManager.getLogger(TimeHandler.class);
+
+    /**
+     * IRI of the asset.
+     */
+    private final String iri;
+
+    /**
+     * Name of matching class.
+     */
+    private final String classMatch;
+
+    /**
+     * Endpoint(s) for the KG.
+     */
+    private final List<ConfigEndpoint> endpoints = new ArrayList<>();
+
+    /**
+     * Connection to KG.
+     */
+    private RemoteStoreClient rsClient;
+
+    /**
+     * Connection to timeseries RDB
+     */
+    private TimeSeriesClient<Instant> tsClient;
+
+    /**
+     * Last n hours to get timeseries data for.
+     */
+    private int hours = 24;
+
+    /**
+     * Response to write details back to.
+     */
+    private HttpServletResponse response;
+
+    /**
+     * Persistent connection to the underlying RDB.
+     */
+    private Connection rdbConnection;
+
+    /**
+     * Initialise a new TimeHandler.
+     * 
+     * @param iri IRI of the asset.
+     * @param classMatch name of class for asset.
+     * @param rdbConnection Persistent connection to the underlying RDB.
+     * @param endpoint Blazegraph endpoint for the KG.
+     */
+    public TimeHandler(String iri, String classMatch, Connection rdbConnection, ConfigEndpoint endpoint) {
+        this(iri, classMatch, rdbConnection, Arrays.asList(endpoint));
+    }
+
+    /**
+     * Initialise a new TimeHandler with multiple endpoints.
+     * 
+     * @param iri IRI of the asset.
+     * @param classMatch name of class for asset
+     * @param rdbConection Persistent connection to the underlying RDB.
+     * @param endpoints Blazegraph endpoints for the KG.
+     */
+    public TimeHandler(String iri, String classMatch, Connection rdbConnection, List<ConfigEndpoint> endpoints) {
+        String fixedIRI = iri;
+        if(!fixedIRI.startsWith("<")) fixedIRI = "<" + fixedIRI;
+        if(!fixedIRI.endsWith(">")) fixedIRI = fixedIRI + ">";
+
+        this.iri = fixedIRI;
+        this.classMatch = classMatch;
+        this.rdbConnection = rdbConnection;
```

Comment:
It looks like you are storing the connection object, it's not recommended, you should do something like
`try (Connection conn = rdbClient.getConnection()) {
    timeSerieClient.getTimeSeries(....., conn)
}`